unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BuildModeHUD/Scripts/Common/SceneCatalogController.cs

```diff
@@ -0,0 +1,374 @@
+using DCL.Configuration;
+using DCL.Helpers;
+using System;
+using System.Collections.Generic;
+
+public interface ISceneCatalogController
+{
+    event Action OnHideCatalogClicked;
+    event Action<CatalogItem> OnCatalogItemSelected;
+    event Action OnResumeInput;
+    event Action OnStopInput;
+
+    void Initialize(ISceneCatalogView view, IQuickBarController quickBarController);
+    void Dispose();
+    void AssetsPackFilter(bool isOn);
+    void CategoryFilter(bool isOn);
+    void FavoritesFilter(bool isOn);
+    void ToggleCatalogExpanse();
+    void OnSearchInputChanged(string currentSearchInput);
+    List<Dictionary<string, List<CatalogItem>>> FilterAssets(string nameToFilter);
+    void QuickBarInput(int quickBarSlot);
+    void ShowFavorites();
+    void CatalogItemSelected(CatalogItem catalogItem);
+    void OnCatalogItemPackSelected(CatalogItemPack catalogItemPack);
+    void SceneCatalogBack();
+    bool IsCatalogOpen();
+    void ShowCategories();
+    void ShowAssetsPacks();
+    void ShowCatalogContent();
+    void OpenCatalog();
+    void CloseCatalog();
+    void RefreshAssetPack();
+    void RefreshCatalog();
+    CatalogItemAdapter GetLastCatalogItemDragged();
+}
+
+public class SceneCatalogController : ISceneCatalogController
+{
+    public event Action OnHideCatalogClicked;
+    public event Action<CatalogItem> OnCatalogItemSelected;
+    public event Action OnResumeInput;
+    public event Action OnStopInput;
+
+    internal ISceneCatalogView sceneCatalogView;
+    internal IQuickBarController quickBarController;
+    private FavoritesController favoritesController;
+    internal List<Dictionary<string, List<CatalogItem>>> filterObjects = new List<Dictionary<string, List<CatalogItem>>>();
+    internal bool isShowingAssetPacks = false;
+    private bool isFilterByAssetPacks = true;
+    internal const string FAVORITE_NAME = "Favorites";
```

Comment:
can we move the constant to the top of the class definition? sory I have that obsession :P

Reply:
I totally understand these kind of obsessions, I feel exactly the same way :) Done in https://github.com/decentraland/explorer/pull/2103/commits/66ed74b073b7d18de5a29f3fbcace90ddfcf293c